### DISCUSSION TITLE
How to trigger a toast during page navigation (NextJS)

### DISCUSSION THREAD

**Discussion by veloware** [chakra-ui/chakra-ui#9274] [Q&A] [answered]
Is it possible to trigger a toast and navigate at the same time in NextJS, so that the toast remains on the screen?

If tried to putting the `<Toaster />` component in the root layout as it is used across all pages, then on success of an API call im doing - 

```
 onSuccess: () => {
      toaster.create({
        title: "Application Submitted",
        description:
          "Your request to become an organizer has been sent successfully.",
        type: "success",
        duration: 5000,
      });
      router.push(ORGANIZER);
    },
```

But the toast doesn't appear.  If I remove the router.push(), the toast appears.  I have also tried to handle the navigation in the server action using `redirect()` but same thing.

**Comment by segunadebayo** [comment, marked as answer]
Keep `<Toaster />` in a layout that does not remount. Call `toaster.create`, then navigate.
